### PR TITLE
Added tests for storing float, boolean, array and object values

### PR DESCRIPTION
**Pull request by kynx**
Hi @Nyholm - a few more tests to round out storing different value types.

### CODE REVIEW

File: src/CachePoolTest.php [hunk 1]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+        $float = 1.23456789;
+        $item = $this->cache->getItem('key');
+        $item->set($float);
+        $this->cache->save($item);
+
+        $this->assertTrue($this->cache->hasItem('key'), 'Null is a perfectly fine cache value. hasItem() should return true when null are stored. ');
+        $item = $this->cache->getItem('key');
+        $this->assertTrue(is_float($item->get()), 'Wrong data type. If we store float we must get an float back.');
+        $this->assertEquals($float, $item->get());
+        $this->assertTrue($item->isHit(), 'Null is a perfectly fine cache value. isHit() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php:622]: This test could be removed. It was only for data type `null`

File: src/CachePoolTest.php [hunk 2]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+
+            return;
+        }
+
+        $float = 1.23456789;
+        $item = $this->cache->getItem('key');
+        $item->set($float);
+        $this->cache->save($item);
+
+        $this->assertTrue($this->cache->hasItem('key'), 'Null is a perfectly fine cache value. hasItem() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php]: This test could be removed. It was only for data type `null`

**Comment by Nyholm** [comment]
Awesome, Thank you!.

Each test function got two test to many. I made some comments about it.

File: src/CachePoolTest.php [hunk 3]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+            $this->markTestSkipped($this->skippedTests[__FUNCTION__]);
+
+            return;
+        }
+
+        $item = $this->cache->getItem('key');
+        $item->set(true);
+        $this->cache->save($item);
+
+        $this->assertTrue($this->cache->hasItem('key'), 'Null is a perfectly fine cache value. hasItem() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php]: Same here

File: src/CachePoolTest.php [hunk 4]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+        $item = $this->cache->getItem('key');
+        $this->assertTrue(is_bool($item->get()), 'Wrong data type. If we store boolean we must get an boolean back.');
+        $this->assertTrue($item->get());
+        $this->assertTrue($item->isHit(), 'Null is a perfectly fine cache value. isHit() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php:640]: Same here

File: src/CachePoolTest.php [hunk 5]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+
+            return;
+        }
+
+        $array = ['a' => 'foo', 2 => 'bar'];
+        $item = $this->cache->getItem('key');
+        $item->set($array);
+        $this->cache->save($item);
+
+        $this->assertTrue($this->cache->hasItem('key'), 'Null is a perfectly fine cache value. hasItem() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php]: Same here

File: src/CachePoolTest.php [hunk 6]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+        $item = $this->cache->getItem('key');
+        $this->assertTrue(is_array($item->get()), 'Wrong data type. If we store array we must get an array back.');
+        $this->assertEquals($array, $item->get());
+        $this->assertTrue($item->isHit(), 'Null is a perfectly fine cache value. isHit() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php:659]: Same here

File: src/CachePoolTest.php [hunk 7]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+            return;
+        }
+
+        $object = new \stdClass();
+        $object->a = 'foo';
+        $item = $this->cache->getItem('key');
+        $item->set($object);
+        $this->cache->save($item);
+
+        $this->assertTrue($this->cache->hasItem('key'), 'Null is a perfectly fine cache value. hasItem() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php]: Same here

File: src/CachePoolTest.php [hunk 8]
```diff
@@ -603,6 +603,86 @@ public function testDataTypeNull()
...
+        $item = $this->cache->getItem('key');
+        $this->assertTrue(is_object($item->get()), 'Wrong data type. If we store array we must get an array back.');
+        $this->assertEquals($object, $item->get());
+        $this->assertTrue($item->isHit(), 'Null is a perfectly fine cache value. isHit() should return true when null are stored. ');
```
Comment (Nyholm) [inline src/CachePoolTest.php:679]: Same here

**Comment by Nyholm** [comment]
Thanks, Just fix the style issue and I'm happy to merge

**Comment by kynx** [comment]
Phew - haven't had to line up equals signs in ages! Nothing wrong with that, but which CS is that - just out of interest?

**Comment by Nyholm** [comment]
It is compliant with PSR-2. I know that @aequasi likes it so we went with it.

**Comment by Nyholm** [comment]
Thank you for this PR.

**Comment by cryptiklemur** [comment]
@kynx yeah, it's not part of any "CS", but its a preference of mine

**Pull request merged**